pkg/syncer/pv_to_backingdiskobjectid_mapping.go

```diff
@@ -86,6 +86,10 @@ func csiGetPVtoBackingDiskObjectIdMapping(ctx context.Context, k8sclient clients
 
 	for _, vol := range queryAllResult.Volumes {
 		// NOTE: BackingDiskObjectId is the id of vvol or vSan; BackingDiskId is the same as VolumeId.
+		// This is only supported for block volumes.
+		if vol.VolumeType != string(cnstypes.CnsVolumeTypeBlock) {
+			continue
+		}
```

Comment:
I think it'd be better to do type assertion in addition to your newly introduced checks, at least it would prevent panics.

```
val, ok :=vol.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
if ok {
     volumeIdToBackingObjectIdMap[vol.VolumeId.Id] = val.BackingDiskObjectId
}
```
Refer https://golangdocs.com/type-assertions-in-golang

Reply:
Added type assertion in addition to volume type check.